Fix compilation on Alpine Linux

In file included from /usr/src/tarantool/src/box/xlog.cc:31:0:
/usr/src/tarantool/src/box/xlog.h:110:2: error: 'mode_t' does not name a type
  mode_t mode;
  ^
/usr/src/tarantool/src/box/xlog.cc: In function 'void xdir_create(xdir*, const char*, xdir_type, const tt_uuid*)':
/usr/src/tarantool/src/box/xlog.cc:109:7: error: 'struct xdir' has no member named 'mode'
  dir->mode = 0660;
       ^

--- a/src/box/xlog.h
+++ b/src/box/xlog.h
@@ -32,6 +32,7 @@
  */
 #include <stdio.h>
 #include <stdbool.h>
+#include <sys/stat.h>
 #include "tt_uuid.h"
 #include "vclock.h"
 

